File: Sorting/Bubble Sort/Julia/bubble-sort.jl
# Bubble Sort is the simplest sorting algorithm that works by 
# repeatedly swapping the adjacent elements if they are in wrong order.

function swap!{T}(x::AbstractArray{T}, a::T, b::T)

  tmp = x[a]
  x[a] = x[b]
  x[b] = tmp
  return x

end

function bubblesort!{T}(x::AbstractArray{T})

  for i in 2:length(x)
    for j in 1:length(x)-1
      if x[j] > x[j+1]
        swap!(x, j, j+1)
      end
    end
  end

  return x
end

x = [10,2,5,1,3,0]

println(bubblesort!(x))

File: bubblesort.cpp
#include<iostream>
using namespace std;
int main()  //Bubble Sort
{
    int a[7]={22,1,4,77,33,11,8},N=7,l=0;
    void swap(int*,int*);
    cout<<"Your array:\n";
    for(int i=0;i<N;i++)
        cout<<a[i]<<"\t";
    cout<<endl;
    cout<<"After Bubble sort\n";
    for(int i=0;i<N;i++)
    {
        l=0;
        for(int j=l;j<N-1;j++)
        {
            if(a[j]>a[j+1]){
                swap(&a[j],&a[j+1]);
            }
            l++;
        }
    }

    cout<<"\nUpdated array is:\n";
    for(int i=0;i<N;i++)
        cout<<a[i]<<"\t";

    return 0;
}
void swap(int *a,int *b )
{
    int c;
    c=*a;
    *a=*b;
    *b=c;
}
